sdk/cosmos/azure-cosmos-spark_3-1_2-12/src/main/scala/com/azure/cosmos/spark/CosmosCatalog.scala

```diff
@@ -537,19 +560,72 @@ class CosmosCatalog
           None,
           s"CosmosCatalog(name $catalogName).tryGetContainerMetadata($databaseName, $containerName)"))
           .to(cosmosClientCacheItem =>
-            cosmosClientCacheItem
-              .client
-              .getDatabase(databaseName)
-              .getContainer(containerName)
-              .read()
-              .block()
-              .getProperties
-        ))
+            (
+              cosmosClientCacheItem
+                .client
+                .getDatabase(databaseName)
+                .getContainer(containerName)
+                .read()
+                .block()
+                .getProperties,
+
+              cosmosClientCacheItem
+                .client
+                .getDatabase(databaseName)
+                .getContainer(containerName)
+                .getFeedRanges
+                .block()
+                .asScala
+                .toList,
+
+              try {
+                Some(
+                  (
+                    cosmosClientCacheItem
+                      .client
+                      .getDatabase(databaseName)
+                      .getContainer(containerName)
+                      .readThroughput()
+                      .block()
+                      .getProperties,
+                    false
+                  ))
+              } catch {
+                case error: CosmosException => {
+                  if (error.getStatusCode != 400) {
```

Comment:
why 400 is a special case and we eat 400 error?

Reply:
If the account is Serverless, the backend returns HTTP 400 when you try to read offers (not the best contract but it is what it is)

Reply:
Correct - the REST contarct for offers is just bad. 400 can mean no container throughput but shared database throughput or serverless (or something else). No substatus code to indicate. I have intentionally left this as Unknown and not Serverless because I hope for serverless eventually we will get a proper contract

Reply:
that is very bad contract. :/